### PR TITLE
Fixed TreeBuilder

### PR DESCRIPTION
**Pull request by thomaspicquet**

### CODE REVIEW

**Comment by thomaspicquet** [comment]
Let me know if there's something else you would like me to change ;)

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -18,8 +18,13 @@ class Configuration implements ConfigurationInterface
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('m6_statsd');
+        if (method_exists(new TreeBuilder(), 'getRootNode')) {
```
Comment (Oliboy50) [inline src/DependencyInjection/Configuration.php]:
```suggestion
```

it's ok if we drop support for older versions

if you could just add this line to the `require` key in composer.json:

 ```json
"symfony/config": "^4.2"
```

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -36,17 +41,7 @@ private function addServersSection($rootNode)
...
-                                ->validate()
-                                    ->ifTrue(
-                                        function ($v) {
-                                            return substr($v, 0, 6) != 'udp://';
-                                        }
-                                    )
-                                    ->thenInvalid("address parameter should begin with 'udp://'")
-                                ->end()
-                            ->end()
+                            ->scalarNode('address')->isRequired()->end()
```
Comment (Oliboy50) [inline src/DependencyInjection/Configuration.php]: I'm not sure why you've changed this validation, but it shouldn't hurt anyway 🤷‍♂

Comment (thomaspicquet) [reply src/DependencyInjection/Configuration.php]: for some reason, when I set it up like this, the validation fails:
```
m6_statsd:
    servers:
        default:
            address: '%env(STATSD_SERVER)%'
            port:    '%env(int:STATSD_PORT)%'
    clients:
        default:
            servers: ['default']
```
even tho those are setup correctly with `udp://` in my env

**Pull request closed without merging**